From 765325fd882e4f7ec61b60255276236e1339c884 Mon Sep 17 00:00:00 2001
From: Shubhrajyoti Datta <shubhrajyoti.datta@xilinx.com>
Date: Mon, 5 Jun 2017 10:16:26 +0530
Subject: [PATCH 130/532] misc: trafgen: Fix the clearing of ram

commit 735c85f887f92243f1f9efa3f1073e53c8e342e2 from
https://github.com/Xilinx/linux-xlnx.git

The switch case is failing as we are passing write along with
clear. Fix the same. While at it remove an unnecessary check.

Signed-off-by: Shubhrajyoti Datta <shubhrajyoti.datta@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/misc/xilinx_trafgen.c | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/drivers/misc/xilinx_trafgen.c b/drivers/misc/xilinx_trafgen.c
index 8ab6a71..83aa37e 100644
--- a/drivers/misc/xilinx_trafgen.c
+++ b/drivers/misc/xilinx_trafgen.c
@@ -594,23 +594,20 @@ static ssize_t xtg_sysfs_ioctl(struct device *dev, const char *buf,
 		break;
 
 	case XTG_CLEAR_MRAM:
-		if (wrval)
-			xtg_access_rams(tg, tg->xtg_mram_offset,
-				XTG_MASTER_RAM_SIZE, XTG_WRITE_RAM |
+		xtg_access_rams(tg, tg->xtg_mram_offset,
+				XTG_MASTER_RAM_SIZE,
 				XTG_WRITE_RAM_ZERO, NULL);
 		break;
 
 	case XTG_CLEAR_CRAM:
-		if (wrval)
-			xtg_access_rams(tg, XTG_COMMAND_RAM_OFFSET,
-				XTG_COMMAND_RAM_SIZE, XTG_WRITE_RAM |
+		xtg_access_rams(tg, XTG_COMMAND_RAM_OFFSET,
+				XTG_COMMAND_RAM_SIZE,
 				XTG_WRITE_RAM_ZERO, NULL);
 		break;
 
 	case XTG_CLEAR_PRAM:
-		if (wrval)
-			xtg_access_rams(tg, XTG_PARAM_RAM_OFFSET,
-				XTG_PARAM_RAM_SIZE, XTG_WRITE_RAM |
+		xtg_access_rams(tg, XTG_PARAM_RAM_OFFSET,
+				XTG_PARAM_RAM_SIZE,
 				XTG_WRITE_RAM_ZERO, NULL);
 		break;
 
-- 
2.7.4

